### PR TITLE
[libtorrent] Update port to 2.0.5

### LINKED ISSUE #18192: [libtorrent] update version from 1.2.13 to  2.0.3
Library name: libtorrent

New version number: 2.0.3

-> BitTorrent v2 support

### PR DESCRIPTION
**Pull request by pythagoras9**
**Describe the pull request**
Update libtorrent port to the latest version 2.0.5

Which triplets are supported/not supported? Have you updated the CI baseline?
- all, no

Does your PR follow the maintainer guide?
- yes

If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?
- yes

### CODE REVIEW

**Comment by pythagoras9** [comment]
waiting on issue [https://github.com/arvidn/libtorrent/issues/6678](https://github.com/arvidn/libtorrent/issues/6678)

File: ports/libtorrent2/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+
+
+file(COPY ${TRYSIGNAL_SOURCE_PATH}/ DESTINATION ${SOURCE_PATH}/deps/try_signal)
+file(COPY ${ASIO_GNUTLS_SOURCE_PATH}/ DESTINATION ${SOURCE_PATH}/deps/asio-gnutls)
+file(COPY ${LIB_SIMULATOR_SOURCE_PATH}/ DESTINATION ${SOURCE_PATH}/simulation/libsimulator)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/libtorrent2/portfile.cmake:74]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH “${SOURCE_PATH}”
```

File: ports/libtorrent2/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    OPTIONS
+       ${FEATURE_OPTIONS}
+       -Dboost-python-module-name=${_boost-python-module-name}
+       -Dstatic_runtime=${_static_runtime}
+       -DPython3_USE_STATIC_LIBS=ON
+)
+
+vcpkg_install_cmake()
```
Comment (JonLiu1993) [inline ports/libtorrent2/portfile.cmake:82]:
```suggestion
vcpkg_cmake_install()
```

File: ports/libtorrent2/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/LibtorrentRasterbar TARGET_PATH share/LibtorrentRasterbar)
```
Comment (JonLiu1993) [inline ports/libtorrent2/portfile.cmake:84]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME LibtorrentRasterbar CONFIG_PATH lib/cmake/LibtorrentRasterbar)
```

File: ports/libtorrent2/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/libtorrent2/portfile.cmake:87]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/libtorrent2/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+# Do not duplicate include files
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include ${CURRENT_PACKAGES_DIR}/debug/share ${CURRENT_PACKAGES_DIR}/share/cmake)
```
Comment (JonLiu1993) [inline ports/libtorrent2/portfile.cmake:90]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share" "${CURRENT_PACKAGES_DIR}/share/cmake")
```

File: ports/libtorrent2/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        "boost-python"
+      ]
+    },
+    "test": {
+      "description": "build the libtorrent tests"
+    },
+    "tools": {
+      "description": "build the tools in the tools directory"
+    }
+  }
```
Comment (JonLiu1993) [inline ports/libtorrent2/vcpkg.json:54]:
```suggestion
    "tools": {
      "description": "build the tools in the tools directory"
    },
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
  }
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libtorrent2/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    endif()
+endif()
+
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        deprfun     deprecated-functions
+        examples    build_examples
+	    iconv       iconv
+        python      python-bindings
```
[the remark this answers is not in the document]
Comment (pythagoras9) [reply ports/libtorrent2/portfile.cmake]: done

File: ports/libtorrent2/portfile.cmake (hunk 2 above)
Comment (pythagoras9) [reply ports/libtorrent2/portfile.cmake]: done

File: ports/libtorrent2/portfile.cmake (hunk 1 above)
Comment (pythagoras9) [reply ports/libtorrent2/portfile.cmake]: done

File: ports/libtorrent2/portfile.cmake (hunk 3 above)
Comment (pythagoras9) [reply ports/libtorrent2/portfile.cmake]: done

File: ports/libtorrent2/portfile.cmake (hunk 4 above)
Comment (pythagoras9) [reply ports/libtorrent2/portfile.cmake]: done

File: ports/libtorrent2/portfile.cmake (hunk 5 above)
Comment (pythagoras9) [reply ports/libtorrent2/portfile.cmake]: done

File: ports/libtorrent2/vcpkg.json (hunk 6 above)
Comment (pythagoras9) [reply ports/libtorrent2/vcpkg.json:62]: done

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Except feature 'python'，all features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by ras0219-msft** [comment]
Because this cannot be used side-by-side with the 1.x series of libtorrent libraries and because it is intended to be an upgrade (including a provided upgrade guide at http://libtorrent.org/upgrade_to_2.0-ref.html), it is vcpkg policy that we should upgrade the existing `libtorrent` port to the 2.x series. This follows what Debian is doing as well: https://packages.debian.org/source/experimental/libtorrent-rasterbar.

Could you please apply your upgrade to the existing `libtorrent` instead of adding `libtorrent2`?

**Review by JonLiu1993 — APPROVED** [review]

File: ports/libtorrent/portfile.cmake [hunk 8]
```diff
@@ -19,42 +19,72 @@ vcpkg_check_features(
...
-    SHA512 7be169625c9ab1319047aa025abb36579cf6c749faeee7788b5c0ca24ca32d0a4bfd898415159b9133302da4d26230edcb5585c12137edcab8f5b3548c67a268
-    HEAD_REF RC_1_2
+        OUT_SOURCE_PATH SOURCE_PATH
+        REPO arvidn/libtorrent
+        REF 44e285c30f06772e48f515580f961998e1037b7e # v2.0.5
+        SHA512 c7ce747930ab8c1852d0efdc6268d9faa39643abd6b6560b4b5e0b06a12ab207ec61ac301a0bcc7622521a8ae490cc02cbe0b6e43208bc216c08bf472b40cb85
+        HEAD_REF RC_2_0
+)
+
+vcpkg_from_github(
```
Comment (JackBoosY) [inline ports/libtorrent/portfile.cmake:38]: Are these 3 library required by libtorrent? Are they exist in vcpkg?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libtorrent/portfile.cmake (hunk 8 above)
Comment (pythagoras9) [reply ports/libtorrent/portfile.cmake:38]: yes, they are present as submodules in the repo. They are required. They aren't present in vcpkg

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks, this LGTM!

**Pull request merged**